Remove #if/def directives for UFE_PREVIEW_VERSION_NUM >= 2021 now that PR 119 is out.

HamedSabri-adsk opened the pull request:
Remove #if/def directives for UFE_PREVIEW_VERSION_NUM >= 2021

ppt-adsk reviewed (approved):
Thanks for doing this Hamed.

seando-adsk commented on lib/mayaUsd/ufe/UsdSceneItem.h:
```diff
@@ -48,9 +48,9 @@ class MAYAUSD_CORE_PUBLIC UsdSceneItem : public Ufe::SceneItem
 	// Ufe::SceneItem overrides
 	std::string nodeType() const override;
-#if UFE_PREVIEW_VERSION_NUM >= 2020
+	#if UFE_PREVIEW_VERSION_NUM >= 2020
```
Why didn't you remove this one?

seando-adsk replied:
This should be changed into the UFE_V2 check.

seando-adsk reviewed (changes requested):
Great job thanks. One minor comment.

HamedSabri-adsk commented:
@ppt-adsk @seando-adsk 

Per our offline discussion,  we are unfortunately stuck with some of these #if/#defs since we still need to support UFE 1.0. :(

For now, I changed my #if/#defs from `UFE_PREVIEW_VERSION_NUM >= 2020` to `UFE_V2_FEATURES_AVAILABLE` so at least it 
is a little bit more readable :) 

I also cleaned up some bunch of unnecessary #if/defs. Will add github comments in the PR where it is appropriate tomorrow.

See 346c6f58271a37827f7b8705d10e2e6e885bdb2f
